src/modules/todoList.js

```diff
@@ -0,0 +1,213 @@
+import isStorageAvailable from './isStorageAvailable.js';
+
+export default class TodoList {
+  constructor(wrapper, storageName = 'todos') {
+    this.wrapper = wrapper;
+    this.storageName = storageName;
+    this.isStorageAvailable = isStorageAvailable('localStorage');
+    this.init();
+  }
+
+  init() {
+    if (this.isStorageAvailable) {
+      const storage = window.localStorage.getItem(this.storageName);
+      this.todos = JSON.parse(storage) || [];
+      this.todos = this.sortList();
+      window.localStorage.setItem(this.storageName, JSON.stringify(this.todos));
+    } else {
+      this.todos = [];
+    }
+
+    this.addAllToPage();
+  }
+
+  updateStorage() {
+    if (this.isStorageAvailable) {
+      window.localStorage.setItem(this.storageName, JSON.stringify(this.todos));
+    }
+  }
+
+  sortList() {
+    let sortedTodos = this.todos.sort((a, b) => a.index - b.index);
+    // recreate indexes from 0
+    sortedTodos = sortedTodos.map((todo, index) => ({
+      index,
+      description: todo.description,
+      completed: todo.completed,
+    }));
+    return sortedTodos;
+  }
+
+  addAllToPage() {
+    this.todos.forEach((todo) => {
+      this.addToPage(todo);
+    });
+  }
+
+  removeAllFromPage() {
+    this.wrapper.innerHTML = '';
+  }
+
+  addToPage(todo) {
```

Comment:
Optional, I think you could have created a separated js file containing the addToPage function that will make your code look better and not overwhelming